front-end/src/Views/EditProjectPage.js

```diff
@@ -0,0 +1,395 @@
+    return url;
+  }
+  fetchRelatedEvents = (callback) => {
+    const { projectType } = this.state
+    let type = ""
+    if(projectType === projectTypes.conference){ type = eventTypes.conference }
+    else if (projectType === projectTypes.institution){ type = eventTypes.institution }
+    else if (projectType === projectTypes.journal){ type = eventTypes.journal }
+
+    if(type === ""){
+      return;
+    }
+    const event_filter = { event_type : type };
+    axios.get(`${config.API_URL}${config.Event_Creation_Url}?event_type=${event_filter.event_type}`, event_filter, 
+      { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+      .then(res => {
+        console.log(res.data);
+        const events = res.data;
+        this.setState({events : events}, () =>{
+          if(callback) callback();
+        } );
+
+      })
+  }
+  renderEvents = () => {
+    const { projectType, events, event } = this.state
+    let errorMsg = ""
+    if(projectType === "")
+      errorMsg = "Please provide a project type.";
+    else
+      errorMsg = "No such events at the moment"
+    return (
+      <FormControl>
+        <InputLabel style={{ marginLeft: "12px" }} id="events">Events</InputLabel>
+        <Select
+          style={dropdownMenuStyle}
+          value={event}
+          onChange={this.handleEventChange}
+          labelId="events"
+        >
+          {events.length !== 0
+            ?
+            Object.keys(events).map((id, i) => (
+              <MenuItem value={`${config.API_URL}${config.Event_Creation_Url}${id}/`}>{events[i]["title"]}</MenuItem>
+            ))
+            :
+            <MenuItem disabled value="">{errorMsg}</MenuItem>
+          }
+        </Select>
+      </FormControl>
+    )
+  }
+
+  submitProject = () => {
+    const { projectTitle, projectDescription, projectRequirements, collaborators,
+      isPublic, projectState, projectType, dueDate, event } = this.state;      /* members [] for now */
+    
+    if (projectTitle === "" || projectDescription === "" || projectRequirements === "" || collaborators === []
+      ) {
+      this.setState({ message: Messages.emptyFieldError, messageType: AlertTypes.Warning }, () => {
+        this.handleSnackbarOpen();
+      });
+      return;
+    }
+    const project = {
+      name: projectTitle,
+      description: projectDescription,
+      requirements: projectRequirements,
+      members: collaborators,
+      is_public: isPublic,
+      // state: projectState,
+      // project_type: projectType,
+      due_date: dueDate
+    };
+    if(projectType !== "" )
+      project.project_type = projectType;
+    if(event !== "")
+      project.events = [event]
+    if (projectState !== "")
+      project.state = projectState
+
+    axios.post(`${config.API_URL}${config.Edit_Project_Url}`, project, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
```

Comment:
I think we need to use PUT or PATCH request to edit. 
If you aren't going to change everything, patch request may be more appropriate.